internal/trace/url.go

```diff
@@ -1,48 +1,14 @@
 package trace
 
-import (
-	"strings"
-	"sync"
-	"text/template"
-
-	"github.com/sourcegraph/sourcegraph/internal/conf/conftypes"
-)
-
-var (
-	cachedURLTemplateStr string
-	cachedURLTemplate    *template.Template
-	cachedURLTemplateErr error
-	cachedURLTemplateMu  sync.Mutex
-)
-
-// URL returns a trace URL for the given trace ID at the given external URL.
-func URL(traceID string, querier conftypes.SiteConfigQuerier) string {
+// URL returns a trace URL for the given trace ID using the configured trace URL
+// renderer. See trace.Init for more information.
+func URL(traceID string) string {
 	if traceID == "" {
 		return ""
 	}
-	c := querier.SiteConfig()
-	tracing := c.ObservabilityTracing
-	if tracing == nil || tracing.UrlTemplate == "" {
-		return ""
-	}
-
-	cachedURLTemplateMu.Lock()
-	defer cachedURLTemplateMu.Unlock()
-
-	if cachedURLTemplateStr != tracing.UrlTemplate {
-		cachedURLTemplateStr = tracing.UrlTemplate
-		cachedURLTemplate, cachedURLTemplateErr = template.New("traceURL").Parse(tracing.UrlTemplate)
-	}
-
-	if cachedURLTemplateErr != nil {
-		// We contribute a validator on tracer package init, so safe to no-op here
-		return ""
+	if urlRenderer == nil {
+		return "<url renderer not configured see trace.Init>"
```

Comment:
I think giving a more complete path to the undefined variable should be sufficient for someone to trace down what went wrong:

```suggestion
		return "<internal/trace.urlRenderer not configured>"
```